Penteract/Assets/Scripts/FloorIsLava.h

```diff
@@ -0,0 +1,77 @@
+#pragma once
+
+#include "Scripting/Script.h"
+
+#define CORRIDOR_TILES 28
+#define ARENA_TILES 32
+
+class FloorIsLava : public Script
+{
+	GENERATE_BODY(FloorIsLava);
+
+public:
+
+	void Start() override;
+	void Update() override;
+
+public:
+	UID corridorUID = 0;
+	UID arenaUID = 0;
+
+	float timeTilesActive = 10.f;
+	float timeWarning = 3.f;
+
+private:
+
+	GameObject* corridor = nullptr;
+	GameObject* arena = nullptr;
+
+	GameObject* corridorTiles[CORRIDOR_TILES];
+	GameObject* arenaTiles[ARENA_TILES];
+
+	int previousPattern = 1;
+	int currentPattern = 1;
+
+	bool* currentCorridorPattern;
+	bool corridorPattern1[CORRIDOR_TILES] = {	true, false, true, false, true, false, false, false, false, true,
```

Comment:
I feel like this is a dangerous and expensive way to do this. I would have to think of a better way to have the booleans stored, but the way the patterns are stored can definetly be generalized. Perhaps, a way to have this class as a generic class would be to have a Map of patterns and have two different GameObjects that control the corridor patterns and the arena patterns, so this would act as a controller of the different regions of the map, in case more patterns or regions are to be added.
Of course, you don't need to do this, but it's always nice to have a broader view for next developments!! :)